include/reffine/ir/loop.h

```diff
@@ -103,6 +103,30 @@ struct Store : public StmtNode {
     void Accept(Visitor&) final;
 };
 
+struct ThreadIdx : public ExprNode {
```

Comment:
btw, these can be upto 3 dimensions. right now, with this definition we can only support a single dimension grid. 

not sure if we need to support more than 1 dim. but just FYI.

Reply:
Yes, I thought the Y/Z dims were mostly only used when operating on matrices/3d blocks though, which I didn't think would be used here.
It would be simple to update for additional dimensions if it becomes necessary down the road. Or I can do it in a follow up when I have a chance if you prefer to just get it done.

Reply:
no. I don't think we need them any time soon.